app/src/main/java/upday/droidconmvvm/DroidconApplication.java

```diff
@@ -0,0 +1,28 @@
+package upday.droidconmvvm;
+
+import android.app.Application;
+import android.support.annotation.NonNull;
+
+import upday.droidconmvvm.datamodel.DataModel;
+import upday.droidconmvvm.datamodel.IDataModel;
+
+public class DroidconApplication extends Application {
+
+    @NonNull
```

Comment:
This NonNull will light up as warning as it's not final.
For the purpose of the simplicity of example, you could just make it final and assign in right away